sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/IngestionResponsePolicy.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using OpenTelemetry.Exporter.AzureMonitor.Models;
+
+namespace OpenTelemetry.Exporter.AzureMonitor
+{
+    internal class IngestionResponsePolicy : HttpPipelineSynchronousPolicy
+    {
+        public override void OnReceivedResponse(HttpMessage message)
+        {
+            base.OnReceivedResponse(message);
+
+            int itemsAccepted;
+
+            if (message.TryGetProperty("TelemetryItems", out var telemetryItems))
+            {
+                itemsAccepted = ParseResponse(message, (IEnumerable<TelemetryItem>)telemetryItems);
+            }
+            else
+            {
+                itemsAccepted = ParseResponse(message);
+            }
+
+            message.SetProperty("ItemsAccepted", itemsAccepted);
+        }
+
+        internal static int ParseResponse(HttpMessage message, IEnumerable<TelemetryItem> telemetryItems)
+        {
+            var httpStatus = message?.Response?.Status;
+            int itemsAccepted = 0;
+
+            switch (httpStatus)
+            {
+                case ResponseStatusCodes.Success:
+                    itemsAccepted = telemetryItems.Count();
+                    break;
+                case ResponseStatusCodes.PartialSuccess:
+                    // Parse retry-after header
+                    // Send Failed Messages To Storage
+                    break;
+                case ResponseStatusCodes.RequestTimeout:
+                case ResponseStatusCodes.ResponseCodeTooManyRequests:
+                case ResponseStatusCodes.ResponseCodeTooManyRequestsOverExtendedTime:
+                    // Parse retry-after header
+                    // Send Messages To Storage
+                    break;
+                case ResponseStatusCodes.InternalServerError:
+                case ResponseStatusCodes.BadGateway:
+                case ResponseStatusCodes.ServiceUnavailable:
+                case ResponseStatusCodes.GatewayTimeout:
+                    // Send Messages To Storage
+                    break;
+                case null: // UnknownNetworkError
+                    // No HttpMessage. Send TelemetryItems To Storage
+                    break;
+                default:
+                    // Log Non-Retriable Status and don't retry or store;
+                    break;
+            }
+
+            return itemsAccepted;
+        }
+
+        internal static int ParseResponse(HttpMessage message)
```

Comment:
nit: I'm not a fan of the duplicated code in the two ParseResponse methods. 
The only difference I'm seeing is how the `Success` case is handled?

Are you expecting greater changes here in the future?
If not, consider combining these two methods.

Reply:
There is a huge difference in implementation. Try to avoid lot of condition check within the method.